lisp/ess-r-mode.el

```diff
@@ -941,6 +942,42 @@ use \"bin/Rterm.exe\"."
 ;;;###autoload
 (add-to-list 'interpreter-mode-alist '("r" . ess-r-mode))
 
+(defcustom ess-r-insert-pipe-behavior '(newline)
+  "A list that determines the behavior of `ess-r-insert-pipe'.
+Currently supported options are `eol', which means go to the end
+of the current line before inserting a pipe and `newline', which
+inserts a newline and indents after inserting the pipe."
+  :group 'ess-R
+  :type '(sexp)
+  :package-version '(ess . "19.04"))
+
+(defun ess-r-insert-pipe (&optional literally)
+  "Insert a pipe %>%.
+The exact behavior is governed by `ess-r-insert-pipe-behavior',
+which see. Avoids inserting multiple pipes in a row. To insert a
+pipe LITERALLY (bypassing `ess-r-insert-pipe-behavior' and
+checking for multiple pipes), use \\[universal-argument].
+
+This command sets a transient keymap so that you may easily
+repeat it. Repeated invocations call `delete-indentation'. So,
+for example, if bound to \\<ess-r-mode-map>\\[ess-r-insert-pipe],
+and `ess-r-insert-pipe-behavior' is (newline), you can press the
+last key twice to delete the newline easily."
+  (interactive "P")
+  (cond (literally (insert "%>%"))
+        ((equal this-command last-command) (delete-indentation))
```

Comment:
This is not quite what I had in mind. C-c RET RET will not work. 

I would like `ess-r-insert-pipe-behavior` to be a list of lists which users can customize as they want. It might contain more than 2 entries. In fact i think renaming into `ess-r-insert-pipe-style` and using explicit keywords :eol-newline-%>%, :eol-%>%. maggritr has other pipes as well. So this could be refactored into a generic cycling inserted of R %ops%.

Reply:
Ah. Well, that's much different than this implementation. I'll close this then.

Reply:
Not really. I just wanted to put an extra commit on top of it. Can you re-add it?